lib/Makefile.include

```diff
@@ -34,6 +34,20 @@ STANDARD_FLAGS ?= -std=c99
 
 all: $(SRCLIBDIR)/$(LIBNAME).a
 
+GENFILES = \
+	$(TOPDIR)/include/libopencm3/$(TARGET)/nvic.h \
+	$(TOPDIR)/include/libopencmsis/$(TARGET)/irqhandlers.h \
+	$(TOPDIR)/lib/$(TARGET)/vector_nvic.c
+
+IRQ_DEFN = $(TOPDIR)/include/libopencm3/$(TARGET)/irq.json
+
+$(GENFILES) : $(IRQ_DEFN)
+	@printf "  GENHDR   $(TARGET) %s\n" '$<';
+	$(Q)cd $(TOPDIR) && ./scripts/irq2nvic_h ./$(subst $(TOPDIR)/,,$<)
+
+# Real dependencies come in via *.d files
+$(OBJS): | $(if $(wildcard $(IRQ_DEFN)), $(GENFILES),)
```

Comment:
pipe deps? is this a game to try and use the most obscure corners of makefiles possible?

Reply:
In principle I can use a regular prerequisite relation here, but the problem is that would be an overapproximation and cause all object files to be rebuilt if any of the generated headers change regardless of them actually using the header or not -- hardly satisfying. So instead I use order-only and let the `*.d`'s do the real work. I'll add a comment about that.

Reply:
would it be easier here to change the irq generator to have an argument for the type of file to make?  it would run more times, but would make it easier to have "simple" "boring" make rules for each file and clean up things with $(RM) instead of having to run the script with "--cleanup" ?